tools/functional-tester/etcd-tester/stresser.go

```diff
@@ -103,11 +115,11 @@ func (s *stresser) Stress() error {
 
 func (s *stresser) Cancel() {
 	s.mu.Lock()
-	cancel, conn := s.cancel, s.conn
+	cancel, conn, wg := s.cancel, s.conn, s.wg
 	s.mu.Unlock()
 	cancel()
-	// TODO: wait for all routines to exit by adding a waitGroup before calling conn.Close()
 	conn.Close()
+	wg.Wait()
```

Comment:
can you follow the todo? close the conn after wg.Wait? then we probably do not have to check a lot of connection errros.